File: dayu/RemoveDuplicatesFromSortedArrayII.java
public class RemoveDuplicatesFromSortedArrayII {
    public int removeDuplicates(int[] nums) {
        int n = nums.length;
        if (n < 3) return n;
        int end = 1;
        for (int i = 2; i < n; i++) {
            if (nums[i] != nums[end-1]) {
                nums[++end] = nums[i];
            }
        }
        return end + 1;
    }
}

File: mu_wang/9_08/Remove Duplicates from Sorted Array II .py
class Solution(object):
    def removeDuplicates(self, A):
        """
        :type nums: List[int]
        :rtype: int
        """
        i,j=0,0
        if len(A)<1:
            return 0
        A[i]=A[j]
        j+=1
        count=1
        while j< len(A):
            if A[i]==A[j]:
                if count!=2:
                    i+=1
                    A[i]=A[j]
                    count+=1
            else:
                i+=1
                A[i]=A[j]
                count =1
            j+=1
        return i+1
                    
        

        